met-web/src/services/engagementMetadataService/index.ts

```diff
@@ -1,32 +1,102 @@
 import http from 'apiManager/httpRequestHandler';
-import { EngagementMetadata } from 'models/engagement';
+import { EngagementMetadata, MetadataTaxonModify, MetadataTaxon } from 'models/engagement';
 import Endpoints from 'apiManager/endpoints';
 import { replaceUrl } from 'helper';
 
-export const getEngagementMetadata = async (engagementId: number): Promise<EngagementMetadata> => {
+export const getEngagementMetadata = async (engagementId: number): Promise<EngagementMetadata[]> => {
     const url = replaceUrl(Endpoints.EngagementMetadata.GET_BY_ENG, 'engagement_id', String(engagementId));
     if (!engagementId || isNaN(Number(engagementId))) {
-        return Promise.reject('Invalid Engagement Id ' + engagementId);
+        return Promise.reject(new Error('Invalid Engagement Id ' + engagementId));
```

Comment:
Just an FYI, when the async keyword is appended to a function, it'll automatically return a promise so explicit calls to `Promise.resolve`, `Promise.reject` shouldn't be necessary. To have the promise resolve, simply return something. To reject, just throw a new Error